frontend/js/authentication.js

```diff
@@ -0,0 +1,33 @@
+(() => {
+    var authentication_token = localStorage.getItem('authentication_token');
+    if(authentication_token == null) authentication_token = 'DEFAULT';
+
+    $(window).on('load', function() {
+        $("#authentication-token").val(authentication_token);
+    })
+
+    // $('#create-authentication-token').on('click', async function(){
+    //     try {
+    //         $('#new-token-message').hide();
+    //         var new_authentication_token = await makeAPICall(`/v1/authentication/new?name=${$("#new-token-name").val()}`);
+    //         $('#new-token-message').show();
+    //         $('#new-token').text(new_authentication_token.data);
+    //     } catch (err) {
+    //         showNotification('Could not create new authentication token', err);
+    //     }
+    // })
+
+    $('#save-authentication-token').on('click', async function(){
+        localStorage.setItem('authentication_token', $("#authentication-token").val());
+        try {
+            $('#token-details-message').hide();
+            var user_data = await makeAPICall('/v1/authentication/data');
+
+            $('#token-details-message').show();
+            $('#token-details').text(JSON.stringify(user_data.data, null, 2));
+        } catch (err) {
+            showNotification('Could not read authentication token data', err);
```

Comment:
I would think about deleting the token here. Otherwise the user might have added a wrong token, got the error and navigated somewhere else and the frontend stops working because the auth keeps failing.

Reply:
Jap, that would be very confusing. Changed it so that the DEFAULT is now only applied in the API. Not in the Users.py class and not in the frontend anymore. This is clearer.
Also submitting an empty string is now disallowed.

I also added a unit test